### PR TITLE
Cleanup dead code

### PR DESCRIPTION
**Pull request by mfojtik**
This PR just cleans up some dead code and fixes small coding nits.

### CODE REVIEW

File: pkg/authorization/registry/selfsubjectrulesreview/storage.go [hunk 1]
```diff
@@ -79,19 +77,3 @@ func (r *REST) Create(ctx apirequest.Context, obj runtime.Object, _ bool) (runti
 
 	return ret, nil
 }
-
-func (r *REST) filterRulesByScopes(rules []rbac.PolicyRule, scopes []string, namespace string) ([]rbac.PolicyRule, error) {
```
Comment (deads2k) [inline pkg/authorization/registry/selfsubjectrulesreview/storage.go]: huh, I wonder how it's working.

File: pkg/build/apis/build/v1/conversion.go [hunk 2]
```diff
@@ -187,10 +183,7 @@ func addLegacyFieldSelectorKeyConversions(scheme *runtime.Scheme) error {
 }
 
 func addFieldSelectorKeyConversions(scheme *runtime.Scheme) error {
-	if err := scheme.AddFieldLabelConversionFunc(SchemeGroupVersion.String(), "Build", buildFieldSelectorKeyConversionFunc); err != nil {
-		return err
-	}
-	return nil
+	return scheme.AddFieldLabelConversionFunc(SchemeGroupVersion.String(), "Build", buildFieldSelectorKeyConversionFunc)
```
Comment (deads2k) [inline pkg/build/apis/build/v1/conversion.go]: not personally a fan.  I guess @soltysh likes it though.

File: pkg/cmd/server/api/helpers.go [hunk 3]
```diff
@@ -398,7 +398,7 @@ func DefaultClientTransport(rt http.RoundTripper) http.RoundTripper {
 		Timeout:   30 * time.Second,
 		KeepAlive: 30 * time.Second,
 	}
-	transport.Dial = dialer.Dial
```
Comment (deads2k) [inline pkg/cmd/server/api/helpers.go]: I don't know exactly what this does.  Remove this particular change and open an issue to figure out why we have our own DefaultClientTransport.

File: pkg/image/controller/scheduled_image_controller.go [hunk 4]
```diff
@@ -99,7 +99,7 @@ func (s *ScheduledImageStreamController) deleteImageStream(obj interface{}) {
 		if !objIsTombstone {
 			return
 		}
-		stream, isStream = tombstone.Obj.(*imageapi.ImageStream)
+		stream, _ = tombstone.Obj.(*imageapi.ImageStream)
```
Comment (deads2k) [inline pkg/image/controller/scheduled_image_controller.go]: normal controllers have another check here.  Bring it in.

File: pkg/oc/admin/policy/review.go [hunk 5]
```diff
@@ -232,7 +232,7 @@ const (
 func (s *sccReviewHumanReadablePrinter) print(info *resource.Info, obj runtime.Object, out io.Writer) error {
 	w := tabwriter.NewWriter(out, sccReviewTabWriterMinWidth, sccReviewTabWriterWidth, sccReviewTabWriterPadding, sccReviewTabWriterPadChar, sccReviewTabWriterFlags)
 	defer w.Flush()
-	if s.noHeaders == false {
```
Comment (deads2k) [inline pkg/oc/admin/policy/review.go]: prefer this version for a negative of a negative.

Comment (deads2k) [reply pkg/oc/admin/policy/review.go]: I could also accept a bool `needsHeaders := !s.noHeaders`

**Comment by deads2k** [comment]
Looks like some of this cleanup isn't functionally equivalent.  Broke impersonation.

File: pkg/cmd/server/handlers/impersonation_scopes.go [hunk 6]
```diff
@@ -12,10 +12,8 @@ import (
 // TranslateLegacyScopeImpersonation is a filter that will translates user scope impersonation for openshift into the equivalent kube headers.
 func TranslateLegacyScopeImpersonation(handler http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		for _, scope := range req.Header[authenticationapi.ImpersonateUserScopeHeader] {
-			req.Header[authenticationv1.ImpersonateUserExtraHeaderPrefix+authorizationapi.ScopesKey] =
-				append(req.Header[authenticationv1.ImpersonateUserExtraHeaderPrefix+authorizationapi.ScopesKey], scope)
-		}
+		req.Header[authenticationv1.ImpersonateUserExtraHeaderPrefix+authorizationapi.ScopesKey] =
```
Comment (deads2k) [inline pkg/cmd/server/handlers/impersonation_scopes.go]: This is the one that broke us.  We went from not setting it to setting an empty slice.  Can you tag every one of these transformations for review or simply revert them?

File: pkg/authorization/registry/selfsubjectrulesreview/storage.go (hunk 1 above)
Comment (mfojtik) [reply pkg/authorization/registry/selfsubjectrulesreview/storage.go]: @deads2k kind find any reference of where this func is used...

**Comment by mfojtik** [comment]
@kargakis it seems like the verify is missing the link :(

File: pkg/build/admission/config.go [hunk 7]
```diff
@@ -14,8 +14,8 @@ import (
 func ReadPluginConfig(pluginConfig map[string]configapi.AdmissionPluginConfig, name string, config runtime.Object) error {
 
 	configFilePath, err := pluginconfig.GetPluginConfigFile(pluginConfig, name, "")
```
Comment (soltysh) [inline pkg/build/admission/config.go]: OMG, ain't nothing better than swallowed error 😞  Good catch!

File: pkg/build/builder/docker.go [hunk 8]
```diff
@@ -360,10 +360,10 @@ func getDockerfilePath(dir string, build *buildapi.Build) string {
 }
 func parseDockerfile(dockerfilePath string) (*parser.Node, error) {
 	f, err := os.Open(dockerfilePath)
-	defer f.Close()
```
Comment (soltysh) [inline pkg/build/builder/docker.go]: Both are equivalent, although the previous version was more readable. Being able to have the deferred close call next to open is the reason for this construct in golang.

**Pull request merged**